metadata-ingestion/src/datahub/ingestion/source_report/ingestion_stage.py

```diff
@@ -20,31 +23,70 @@
 PROFILING = "Profiling"
 
 
+class IngestionHighStage(Enum):
+    """
+    The high-level stages at the framework level
+    Team to add more stages as needed
+    """
+
+    PROFILING = "Profiling"
+    _UNDEFINED = "Ingestion"
+
+
 @dataclass
 class IngestionStageReport:
-    ingestion_stage_durations: TopKDict[str, float] = field(default_factory=TopKDict)
+    ingestion_high_stage_seconds: dict[str, float] = field(
+        default_factory=lambda: defaultdict(float)
+    )
+    ingestion_stage_durations: TopKDict[Tuple[str, str], float] = field(
+        default_factory=TopKDict
+    )
```

Comment:
may we use the `IngestionHighStage` type instead of the string?

```suggestion
    ingestion_high_stage_seconds: dict[IngestionHighStage, float] = field(
        default_factory=lambda: defaultdict(float)
    )
    ingestion_stage_durations: TopKDict[Tuple[IngestionHighStage, str], float] = field(
        default_factory=TopKDict
    )
```